Packs/AzureNetworkSecurityGroups/Integrations/AzureNetworkSecurityGroups/AzureNetworkSecurityGroups_test.py

```diff
@@ -57,6 +57,115 @@ def test_list_groups_command(mocker):
     assert results.outputs[0].get('name') == 'alerts-nsg'
 
 
+def test_azure_nsg_public_ip_addresses_list(mocker):
+    """
+    Validate that azure_nsg_public_ip_addresses_list returns the output in the correct format
```

Comment:
In all unit test's docstrings, use `Given, When, Then` terminology to describe what the test does.